test/test_datapipe.py

```diff
@@ -248,6 +255,124 @@ def test_groupbykey_iterable_datapipe(self):
+                                  file_url_template):
+    furl_local_file = os.path.join(tmp_dir, "urls_list")
+    with open(furl_local_file, 'w') as fsum:
+        for i in range(0, file_count):
+            f = os.path.join(tmp_dir, "webfile_test_{num}.data".format(num=i))
+
+            write_chunk = 1024 * 1024 * 16
+            rmn_size = file_size
+            while rmn_size > 0:
+                with open(f, 'ab+') as fout:
+                    fout.write(os.urandom(min(rmn_size, write_chunk)))
+                rmn_size = rmn_size - min(rmn_size, write_chunk)
+
+            fsum.write(file_url_template.format(num=i))
+
+
+class TestIterableDataPipeHttp(TestCase):
+    __server_thread: threading.Thread
+    __server_addr: str
+    __server: socketserver.TCPServer
+
+    @classmethod
+    def setUpClass(cls):
+        try:
+            (cls.__server_thread, cls.__server_addr,
+             cls.__server) = setUpLocalServerInThread()
+        except Exception as e:
+            warnings.warn("TestIterableDataPipeHttp could\
+                          not set up due to {0}".format(str(e)))
+
+    @classmethod
+    def tearDownClass(cls):
+        try:
+            cls.__server.shutdown()
+            cls.__server_thread.join(timeout=15)
+        except Exception as e:
+            warnings.warn("TestIterableDataPipeHttp could\
+                           not tear down (clean up temp directory or terminate\
+                           local server) due to {0}".format(str(e)))
+
+    def http_test_base(self, test_file_size, test_file_count, timeout=None,
+                       chunk=None):
+        with tempfile.TemporaryDirectory(dir=os.getcwd()) as tmpdir:
+            # create tmp dir and files for test
+            base_tmp_dir = os.path.basename(os.path.normpath(tmpdir))
+            file_url_template = ("http://{server_addr}/{tmp_dir}/"
+                                 "/webfile_test_{num}.data\n")\
+                .format(server_addr=self.__server_addr, tmp_dir=base_tmp_dir,
+                        num='{num}')
+            create_temp_files_for_serving(tmpdir, test_file_count,
+                                          test_file_size, file_url_template)
+
+            datapipe_dirf = dp.iter.ListDirFiles(tmpdir, '*_list')
+            datapipe_rf = dp.iter.ReadLinesFromFile(datapipe_dirf)
+            datapipe_tup = TupleToElementIDP(datapipe_rf, 1)
+            datapipe_http = dp.iter.HttpReader(datapipe_tup, timeout=timeout)
+            datapipe_tob = dp.iter.ToBytes(datapipe_http, chunk=chunk)
+
+            for (url, data) in datapipe_tob:
+                self.assertGreater(len(url), 0)
+                self.assertRegex(url, r'^http://.+\d+.data$')
+                if chunk is not None:
+                    self.assertEqual(len(data), chunk)
+                else:
+                    self.assertEqual(len(data), test_file_size)
+
+    def test_stress_http_reader_iterable_datapipes(self):
```

Comment:
Please disable this test, as we can't afford to run it in CI env.

Reply:
agree, they could be slow if config to a larger number.  will disable both for now.